### PR TITLE
[@types/tabulator-tables] Update types for `OptionRowGrouping`

### PR DESCRIPTION
**Pull request by ghost**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://tabulator.info/docs/5.5/group#overview
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/tabulator-tables/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 217e0333583e900131ccc717f646e48d654af832
</p>

File: types/tabulator-tables/index.d.ts [hunk 1]
```diff
@@ -319,7 +319,7 @@ export interface OptionsRowGrouping {
      * Group Open Function
      * If you want to decide on a group by group basis which should start open or closed then you can pass a function to the groupStartOpen property. This should return true if the group should start open or false if the group should start closed.
      */
-    groupStartOpen?: boolean | ((value: any, count: number, data: any, group: GroupComponent) => boolean) | undefined;
+    groupStartOpen?: boolean | boolean[] | ((value: any, count: number, data: any, group: GroupComponent) => boolean) | undefined;
```
Comment (ortwic) [inline types/tabulator-tables/index.d.ts:322]: Can be also an array of mixed booleans and funcs.

File: types/tabulator-tables/index.d.ts [hunk 2]
```diff
@@ -285,7 +285,7 @@ export interface OptionsPagination {
     paginationInitialPage?: number | undefined;
 }
 
-export type GroupArg = string | string[] | ((data: any) => any);
+export type GroupArg = string | string[] | ((data: any) => any) | Array<(data: any) => any>;
```
Comment (ortwic) [inline types/tabulator-tables/index.d.ts:288]: Can be a mixed array of string or funcs

File: types/tabulator-tables/index.d.ts [hunk 3]
```diff
@@ -285,7 +285,7 @@ export interface OptionsPagination {
     paginationInitialPage?: number | undefined;
 }
 
-export type GroupArg = string | string[] | ((data: any) => any);
+export type GroupArg = string | string[] | ((data: any) => any) | Array<(data: any) => any> | Array<string | (data: any) => any>;
```
Comment (mike-lischke) [inline types/tabulator-tables/index.d.ts:288]: The second last entry is a duplicate of the more general form in the last entry, so it can go.

**Review by mike-lischke — CHANGES_REQUESTED** [review]
Please add a test for this change.

**Review by mike-lischke — COMMENTED** [review]
Looks good, but can you add a little test, which uses the new definitions?

**Review by mike-lischke — APPROVED** [review]

**Review by mike-lischke — APPROVED** [review]

**Review by mike-lischke — APPROVED** [review]

**Comment by codershiba** [comment]
Ready to merge

**Pull request merged**